lighthouse-core/gather/gatherers/scripts.js

```diff
@@ -20,21 +23,53 @@ class Scripts extends Gatherer {
   async afterPass(passContext, loadData) {
     const driver = passContext.driver;
 
-    /** @type {Object<string, string>} */
-    const scriptContentMap = {};
+    /** @type {LH.Artifacts['Scripts']} */
+    const scripts = [];
+
+    /** @type {string[]} */
+    const inlineScripts = await driver.evaluateAsync(`(() => {
+      ${getElementsInDocumentString};
+
+      return getElementsInDocument('script')
+        .filter(script => !script.src && script.text.trim())
+        .map(script => script.text);
+    })()`, {useIsolation: true});
+
+    if (inlineScripts.length) {
+      const mainResource = loadData.networkRecords.find(request =>
+        passContext.url.startsWith(request.url) &&
+          URL.equalWithExcludedFragments(request.url, passContext.url));
+      if (!mainResource) {
+        log.warn('Scripts', 'could not locate mainResource');
+      }
+      scripts.push(
+        ...inlineScripts.map(content => {
+          return {
+            content,
+            inline: true,
+            requestId: mainResource ? mainResource.requestId : undefined,
```

Comment:
move this out of the `map` to a named variable to keep it cleaner in here.

It definitely sucks that we have to add this for the case that should never happen but turns out does. Maybe add a comment too about `mainResource` is expected to be found, but on the very small chance something weird happens and it isn't found, we'll roll without it for now?

Reply:
oh, actually, forgot about the earlier conversation. We should switch to `NetworkAnalyzer.findMainDocument()` once #7080 lands so we can make `requestId` not optional